acceptance/Gemfile

```diff
@@ -1,11 +1,9 @@
-# Specifies a gem mirror; duplicated in acceptance setup
-# to ensure a similar environment on acceptance hosts.
-source ENV['GEM_SOURCE'] || 'https://artifactory.delivery.puppetlabs.net/artifactory/api/gems/rubygems/'
+source ENV['GEM_SOURCE'] || 'https://rubygems.org'
 
 def location_for(place, fake_version = nil)
-  if place =~ /^((?:git[:@]|https:)[^#]*)#(.*)/
+  if place.is_a?(String) && place =~ /^((?:git[:@]|https:)[^#]*)#(.*)/
```

Comment:
I think we have defaults for these being strings. I've had luck with this definition:

```ruby
def location_for(place, fake_version = nil)
  case place
  when /^(git:[^#]*)#(.*)/
    [fake_version, { git: Regexp.last_match(1), branch: Regexp.last_match(2), require: false }].compact
  when %r{^file://(.*)}
    ['>= 0', { path: File.expand_path(Regexp.last_match(1)), require: false }]
  else
    [place, { require: false }]
  end
end
```

Reply:
Yeah, typically `place` _should_ be a string, though I have run into people using arrays for gem versions ([I ran into this](https://github.com/voxpupuli/beaker-puppet/blob/94be1761ea48d775509a2c3dfe38d41d3ecb6786/Gemfile#L17) last week). I was modeling this off of what we did in Puppet a few years back: https://github.com/puppetlabs/puppet/commit/d84b38898527d2c94c1112402a085729c81a6236

Reply:
Cool, whatever works! thanks for updating these!!1